cci_cameroon/analysis/clustering_rumours/rumour_clustering_template.py

```diff
@@ -0,0 +1,412 @@
+)
+faiss.normalize_L2(sentence_embeddings)  # to be used for inner product computation
+neighbors = 10
+
+# %%
+indp.train(sentence_embeddings)
+indp.add(sentence_embeddings)
+distance_matric, position = indp.search(
+    sentence_embeddings, sentence_embeddings.shape[0]
+)  # create similarity matrix for the data
+
+# %%
+# create a KNN network with n neighbors using the cosine similarity matrix
+A = kneighbors_graph(distance_matric, 5, mode="connectivity", include_self=False)
+A.toarray()[0]
+
+# %%
+# load the graph
+G = nx.from_numpy_matrix(A, create_using=nx.Graph())
+print(G)
+# visualize the graph
+nx.draw(G, with_labels=False)
+
+# %%
+len(G.nodes), len(G.edges)
+
+# %%
+# using leiden community detection algorithm on the dataset
+coms = algorithms.leiden(G)
+draw_communities_graph(G, generate_colors(coms), coms)
+
+# %%
+len(coms.communities)
+
+# %%
+for index in range(len(coms.communities)):
+    print(data.comment[coms.communities[index]].index)
+    print("######################NEW Group####")
+
+# %%
+list(
+    data.iloc[[158, 161, 162, 163, 164, 165, 166, 167, 168, 171]].comment
+)  # belief about mask wearing by children
+
+# %%
+list(
+    data.iloc[
+        [
+            89,
+            92,
+            94,
+            104,
+            108,
+            110,
+            115,
+            116,
+            117,
+            119,
+            120,
+            126,
+            136,
+            139,
+            140,
+            143,
+            153,
+            155,
+            156,
+            157,
+            159,
+            169,
+            173,
+            180,
+            183,
+            184,
+            185,
+            186,
+            187,
+            189,
+            192,
+            193,
+            197,
+            198,
+            199,
+            208,
+            215,
+            216,
+            218,
+            270,
+            291,
+        ]
+    ].comment
+)
+
+# %%
+list(
+    data.iloc[
+        [
+            0,
+            1,
+            3,
+            4,
+            7,
+            13,
+            18,
+            22,
+            23,
+            24,
+            28,
+            31,
+            33,
+            36,
+            38,
+            64,
+            67,
+            75,
+            79,
+            85,
+            145,
+            328,
+            330,
+            331,
+            341,
+            343,
+        ]
+    ].comment
+)
+
+# %%
+list(
+    data.iloc[
+        [
+            2,
+            27,
+            30,
+            32,
+            35,
+            37,
+            54,
+            55,
+            56,
+            57,
+            58,
+            59,
+            60,
+            61,
+            62,
+            63,
+            65,
+            66,
+            80,
+            83,
+            84,
+            338,
+        ]
+    ].comment
+)
+
+# %%
+## Using walktrap community detection algorithm on the data subset
```

Comment:
Is there a reason for testing on such a small subset? How are the results below? (I translated a few but hard to get a sense). I noticed the results seemed less affective using the previous method on the smaller subset compared to the larger dataset - so maybe you need to test on the larger aswell?